torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -405,7 +394,7 @@ void insertQuantizationOps(
     return;
   }
   if (quant_type == QuantType::DYNAMIC) {
-    if (isFP16NoopObserver(module, observer)) {
+    if (isFp16Observer(observer->input(0))) {
```

Comment:
can we just check dtype here?

Reply:
I think so, was just being more explicit by checking for the observer type as well.